Fix double quotation marks causing error

stdout-se opened the pull request:
When parsing a file containing double quotation marks ("), the tokenizer
would throw an error due to not being able to identify the character.

This error has been fixed by allowing the tokenize_code() function to
treat double quotation marks as strings, using the skip_string()
function.

This fixes bastibe/MatlabCodeAnalyzer#10

bastibe commented:
Thank you for your pull request!

Looking at the code, I think this will not handle strings with embedded quotation marks correctly, yet, like `"let's go"` or `'He said, "hi"'`, right?

You could probably implement that by passing `"` or `'` to `skip_string()`.

stdout-se commented:
Good point. I might implement some unit tests for that.

bastibe commented:
I'd be delighted!

stdout-se committed:
Fix double quotation marks causing error

When parsing a file containing double quotation marks ("), the tokenizer
would throw an error due to not being able to identify the character.

This error has been fixed by allowing the tokenize_code() function to
treat double quotation marks as strings, using the skip_string()
function.

This fixes bastibe/MatlabCodeAnalyzer#10

stdout-se committed:
Add unit tests

Adding a unit test suite for tokenize_code(), as well as a unit test
runner script.

stdout-se committed:
Fix bug with nested quotes of different types

Fixed so tokenize_code.m can handle strings like ''let's go" or
'he said, "hi"'. This is achieved by passing the initial character
to skip_string().

stdout-se committed:
Fix bug where checked code can't start with a '

If source_code starts with a single quote char ('),
previous = tokenlist(end); in tokenize_code() will throw an error. This
is solved by first checking if is_first_symbol is true.

stdout-se commented:
There we go, problem solved. Take another look and see what you think.

stdout-se committed:
Removed leftover todo comment

bastibe commented:
Cool! Thank you very much!
